simpledotorg/simple-server
# Bugfix: Experiment notifications not going out as expected

**Story card:** [sc-10920](https://app.shortcut.com/simpledotorg/story/10920)

## Because

Notifications haven't been getting created or sent out in Bnagladesh and India since July 3. This is blocking the Bangladesh SMS experiment, so this is a quick fix. A more long-term fix can be picked up as part of [this card](https://app.shortcut.com/simpledotorg/story/10921/).

## This addresses

When checking for patients to be notified for the day, a timestamp and date are compared. Since the date gets implicitly converted to 00:00 in UTC, this comparison has been failing ever since we wrapped the experiment runner so as to run within the local timezone. This fix saves these dates with the correct date in the UTC format while creating a treatment group membership.

The long term fix is to add a schema migration to convert the `expected_return_date` and `experiment_inclusion_date` fields into dates.

## Test instructions

- Create stale and current experiments, their treatment groups and reminder templates.
- Run conduct_daily on them with any date
- Check that treatment group memberships are being created as expected, and more importantly that notifications are being created.
